app/bundles/FormBundle/Helper/PropertiesAccessor.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\FormBundle\Helper;
+
+use Mautic\FormBundle\Model\FormModel;
+
+class PropertiesAccessor
+{
+    /**
+     * @var FormModel
+     */
+    private $formModel;
+
+    public function __construct(FormModel $formModel)
+    {
+        $this->formModel = $formModel;
+    }
+
+    /**
+     * @return array|mixed
+     */
+    public function getProperties(array $field)
+    {
+        if ($field['type'] === 'country' || (!empty($field['leadField']) && !empty($field['properties']['syncList']))) {
+            return $this->formModel->getContactFieldPropertiesList($field['leadField']);
+        } elseif (!empty($field['properties'])) {
+            return $this->getOptionsListFromProperties($field['properties']);
+        }
+
+        return [];
+    }
+
+    /**
+     * @return array
+     */
+    public function getChoices($options)
+    {
+        $choices = [];
+
+        if (is_array($options) && !isset($options[0]['value'])) {
+            return $options;
+        }
+
+        if (!is_array($options)) {
+            $options = explode('|', $options);
+        }
+
+        foreach ($options as $option) {
+            if (is_array($option)) {
+                if (isset($option['label']) && isset($option['alias'])) {
+                    $choices[$option['alias']] = $option['label'];
```

Comment:
This part as well. If it doesn't have 'value' then it won't get to this foreach (line 49). For what array structure was this built?

Reply:
This should cover fieldsprocessing, when I wanted to build choices of all  selectboxes from form etc. 
But I think it was used in first version, and then I remove this feature.

Reply:
Maybe doesn't exists. In first version I've tried process any properties from form fields, or lead fields. I found a few formats (bools, | divided, arrays). Maybe it was country/state field from leadSync